src/lib/vasi-sync/Cargo.toml

```diff
@@ -23,6 +17,9 @@ rand = "0.8.5"
 [target.'cfg(loom)'.dependencies]
 loom = "0.5"
 
+[target.'cfg(miri)'.dependencies]
+libc = { version ="0.2", default-features = false }
```

Comment:
`cargo nono` detects some transitive uses of `std`, but I don't think it'd detect that particular problem.

The only way I know of would be to build a full `no_std` executable or `.so` with its own panic handler, which would cause the link to fail if `std` ended up getting pulled in transitively. Eventually we'll want to do that with the shim itself, but can't if it's part of the same workspace as everything else unless we configure panics to abort for the whole workspace.

I think ultimately we might need to split out a separate `no_std` workspace.